Content.Server/GameObjects/Components/Botany/PlantHolderComponent.cs

```diff
@@ -0,0 +1,850 @@
+                Health -= healthMod;
+                ImproperPressure = true;
+                if (DrawWarnings)
+                    _updateSpriteAfterUpdate = true;
+            }
+            else
+            {
+                ImproperPressure = false;
+            }
+
+            // Seed ideal temperature.
+            if (MathF.Abs(environment.Temperature - Seed.IdealHeat) > Seed.HeatTolerance)
+            {
+                Health -= healthMod;
+                ImproperHeat = true;
+                if (DrawWarnings)
+                    _updateSpriteAfterUpdate = true;
+            }
+            else
+            {
+                ImproperHeat = false;
+            }
+
+            // Gas production.
+            var exudeCount = Seed.ExudeGasses.Count;
+            if (exudeCount > 0)
+            {
+                foreach (var (gas, amount) in Seed.ExudeGasses)
+                {
+                    environment.AdjustMoles(gas, MathF.Max(1f, MathF.Round((amount * MathF.Round(Seed.Potency)) / exudeCount)));
+                }
+            }
+
+            // Toxin levels beyond the plant's tolerance cause damage.
+            // They are, however, slowly reduced over time.
+            if (Toxins > 0)
+            {
+                var toxinUptake = MathF.Max(1, MathF.Round(Toxins / 10f));
+                if (Toxins > Seed.ToxinsTolerance)
+                {
+                    Health -= toxinUptake;
+                }
+
+                Toxins -= toxinUptake;
+                if (DrawWarnings)
+                    _updateSpriteAfterUpdate = true;
+            }
+
+            // Weed levels.
+            if (PestLevel > 0)
+            {
+                // TODO: Carnivorous plants?
+                if (PestLevel > Seed.PestTolerance)
+                {
+                    Health -= HydroponicsSpeedMultiplier;
+                }
+
+                if (DrawWarnings)
+                    _updateSpriteAfterUpdate = true;
+            }
+
+            // Weed levels.
+            if (WeedLevel > 0)
+            {
+                // TODO: Parasitic plants.
+                if (WeedLevel >= Seed.WeedTolerance)
+                {
+                    Health -= HydroponicsSpeedMultiplier;
+                }
+
+                if (DrawWarnings)
+                    _updateSpriteAfterUpdate = true;
+            }
+
+            if (Age > Seed.Lifespan)
+            {
+                Health -= _random.Next(3, 5) * HydroponicsSpeedMultiplier;
+                if (DrawWarnings)
+                    _updateSpriteAfterUpdate = true;
+            }
+            else if(Age < 0) // Revert back to seed packet!
+            {
+                Seed.SpawnSeedPacket(Owner.Transform.Coordinates);
+                RemovePlant();
+                ForceUpdate = true;
+                Update();
+            }
+
+            CheckHealth();
+
+            if (Harvest && Seed.HarvestRepeat == HarvestType.SelfHarvest)
+                AutoHarvest();
```

Comment:
Shouldn't this be below the harvest check below or do you intentionally want to delay autoharvest to the next Update tick?

Reply:
Indeed, we delay it to the next tick